### PR TITLE
feat(contracts): add transferFrom when increasing reward of vault

### PR DESCRIPTION
**Pull request by sm-stack**
# Description

Added transferFrom function and logic of adding validator and boosted reward at `increaseBalanceWithReward` function.

Must be merged after #351

### CODE REVIEW

File: packages/contracts/contracts/L1/AssetManager.sol [hunk 1]
```diff
@@ -729,6 +737,9 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
         uint128 boostedReward,
         uint128 validatorReward
     ) external onlyValidatorManager {
+        // Distribute the reward from a designated vault to the AssetManager contract.
+        ASSET_TOKEN.transferFrom(VALIDATOR_REWARD_VAULT, address(this), baseReward + boostedReward + validatorReward);
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/AssetManager.sol:741]: Please apply formatting.

Comment (seolaoh) [reply packages/contracts/contracts/L1/AssetManager.sol:741]:
```suggestion
        ASSET_TOKEN.safeTransferFrom(VALIDATOR_REWARD_VAULT, address(this), baseReward + boostedReward + validatorReward);
```

**Comment by sm-stack** [comment]
Just checked this functionality works well with Safe Sepolia account. cc. @seolaoh @0xHansLee

**Review by seolaoh — APPROVED** [review]
LGTM

**Review by 0xHansLee — APPROVED** [review]

**Pull request merged**